core/src/test/java/org/renjin/primitives/IdenticalTest.java

```diff
@@ -66,8 +66,8 @@ public void identical() {
 
   @Test
   public void identicalS4() {
-    topLevelContext.getGlobalEnvironment().setVariable("x", new S4Object());
-    topLevelContext.getGlobalEnvironment().setVariable("y", new S4Object());
+    topLevelContext.getGlobalEnvironment().setVariable(topLevelContext, "x", new S4Object());
```

Comment:
This is quite verbose. Add a `setGlobalVariable() ` method to the base case and replace everywhere.